code/mob/living/critter/ai/sawflyai.dm

```diff
@@ -78,30 +79,32 @@
 
 
 /datum/aiTask/timed/targeted/sawfly_attack/get_targets()
-	. = list()
+	. = list() // don't ask why the list is simply a period
 	var/mob/living/critter/robotic/sawfly/owncritter = holder.owner
 
-	for (var/mob/living/C in view(owncritter,target_range))
-		if(C == owncritter) continue
-
-		if(istype(C, /mob/living/critter/robotic/sawfly)) continue
+	for (var/mob/living/C in viewers(max_dist, owncritter))
 		if (C.health < -50 || !isalive(C)) continue
-		if (C.job == "Security Officer" || C.job == "Head of Security")
-			. = list(C) //found a secoff, just return that
+		if(istype(C, /mob/living/critter/robotic/sawfly)) continue
+
+		if(C.mind?.special_role)
+			if (istraitor(C) || isnukeop(C) || isspythief(C) || isnukeopgunbot(C)) // frens :)
+				if (!(C in owncritter.friends))
+					boutput(C, "<span class='alert'> [owncritter]'s IFF system silently flags you as an ally! </span>")
+					owncritter.friends += C
+				continue
+		if(C.job in list( "Head of Security", "Security Officer", "Nanotrasen Security Consultant")) //hopefully this is cheaper than the OR chain I had before
+			. = list(C) //go get em, tiger
 			return
-		if (C in owncritter.friends) continue
-		if (istraitor(C) || isnukeop(C) || isspythief(C)) // frens :)
-			boutput(C, "<span class='alert'> The sawfly's IFF system silently flags you as an ally! </span>")
-			owncritter.friends += C
-			continue
-		if(istype(C, /mob/living/silicon/ai)) continue // AI cores are tanky and distract nukie sawflies in med/robotics
 		. += C //you passed all the checks it, now you get added to the list for consideration
```

Comment:
Consider capping the max targets. If you go and try to evaluate 50 targets your sawfly is going to be dead before it even takes action. Probably return the list early at 5. Or be aggressive and just get the first target and return that, they are assured to have the lowest scalar distance of any target in range.